edweiss-app/__test__/showtime/LectureScreen.test.tsx

```diff
@@ -209,6 +212,46 @@ describe('LectureScreen Component', () => {
         (useDynamicDocs as jest.Mock).mockReturnValue(mockQuestionData); // Mocking `useDynamicDocs` with minimal question data
     });
 
+    it('updates UI when new questions are added dynamically', () => {
```

Comment:
Hey there. First of congrats 🎉 for achieving about 75% of coverage on your code, that's really great! It was not easy for you as your part is really difficult to test. 
I had bit a difficult time under some part of it. I might have some small recommendation for later improvements:
In order to reduce code duplication, we could leverage the same mock  each time.

For instance:

create this mock as a local variable of the test:
```tsx
const updatedMockQuestionData = [
    ...mockQuestionData, 
    { id: '2', data: { text: 'New Question' } }
];

```

and call it this way:

```tsx
it('updates UI when new questions are added dynamically', () => {
    const { rerender } = render(<LectureScreen />);
    (useDynamicDocs as jest.Mock).mockReturnValueOnce(updatedMockQuestionData);
    rerender(<LectureScreen />);
    expect(screen.getByText('New Question')).toBeTruthy();
});
```
This is just a small fix, feel free to do it.

Reply:
Thank you so much I will do this after I have finished improving the Student questions to a stable version